src/app/onboarding/legal-experience/page.tsx

```diff
@@ -1,24 +1,78 @@
 'use client';
 
-import Button from '@/components/Button';
-import COLORS from '@/styles/colors';
+import { useContext, useState, useEffect } from 'react';
 import { H1 } from '@/styles/text';
 import { OnboardingContext } from '@/utils/OnboardingProvider';
-import { useContext } from 'react';
+import TextInput from '@/components/TextInput';
+import InputDropdown from '@/components/InputDropdown';
+import RadioGroup from '@/components/RadioGroup';
+import { ImmigrationLawExperienceEnum, Profile } from '@/types/schema';
+
+const legalExperienceOptions = new Map<ImmigrationLawExperienceEnum, string>([
+  ['HIGH', 'Multiple cases of immigration law experience'],
+  ['MEDIUM', 'Few cases of immigration law experience'],
+  ['LOW', 'One or no case of immigration law experience'],
+]);
 
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const [barNum, setBarNum] = useState('');
+  const [experience, setExperience] =
+    useState<ImmigrationLawExperienceEnum | null>(null);
+  const [registered, setRegistered] = useState('');
+
+  const validBarNum = () =>
+    !Number.isNaN(parseInt(barNum, 10)) && barNum.length === 6;
+
+  const getErrorText = () => {
+    if (barNum !== '' && !validBarNum()) {
+      return 'Must include attorney bar number';
+    }
+    return '';
+  };
+
+  useEffect(() => {
+    if (validBarNum() && experience !== null && registered !== '') {
+      // update profile
+      const partialProfile: Partial<Profile> = {
+        bar_number: barNum,
+        immigration_law_experience: experience,
+        eoir_registered: registered === 'Yes',
+      };
+      onboarding?.updateProfile(partialProfile);
+      // enable continue
+      onboarding?.setCanContinue(true);
+    } else {
+      onboarding?.setCanContinue(false);
+    }
+  }, [barNum, experience, registered]);
```

Comment:
instead of a `useEffect`, I would recommend making individual `onChange` functions for each input (I've added a separate `onChange` prop to each of them that triggers a side effect)

Reply:
This is so we can populate the data by grabbing values from `onboarding` with `useEffect` -- if we keep the one that we have here, it will result in an infinite recursive loop (since they update values in each others' dependency arrays)